collect_app/src/main/java/org/odk/collect/android/widgets/BarcodeWidgetAnswerView.kt

```diff
@@ -0,0 +1,36 @@
+package org.odk.collect.android.widgets
+
+import android.content.Context
+import android.util.TypedValue
+import android.view.LayoutInflater
+import org.odk.collect.android.databinding.BarcodeWidgetAnswerViewBinding
+
+class BarcodeWidgetAnswerView(
+    context: Context,
+    private val fontSize: Int
+) : WidgetAnswerView(context) {
+    private val binding = BarcodeWidgetAnswerViewBinding.inflate(LayoutInflater.from(context), this, true)
+    private var answer: String? = null
+
+    init {
+        setFontSize()
+    }
+
+    override fun setAnswer(answer: String?) {
+        this.answer = stripInvalidCharacters(answer)
```

Comment:
I'm wondering if this should happen in the widget (probably only in `setData`) given it's not really a display concern and more of a "data processing" thing.

Reply:
Fixed.